### ISSUE TITLE
Add pyhon path UE4 hangs in background

### ISSUE THREAD

**Issue by kaedehao** [20tab/UnrealEnginePython#275]
Hi, I've added customized python path in environment PATH. UE4 will hangs at 73% when launching project and splash will disappear instead of showing the error indicating no python found. UE4 will be still hanging in background which can be closed in task manager.

Wired thing is, as long as I install python with .msi, UE4 will be running like charm (even without adding python path in environment PATH). Seems like it's bound to copied python only. I am aware of python27.dll is necessary in order to run python VM and i'm positive that is not the case.

So what on earth make the difference of installed python using .msi, how can I embed python with my tools so that I can distribute them without having other users install python. (I've tired put both python27 folder and python27.dll in built \binary\win64 path, not working either)

I'm currently developing with python 2.7.14. (I've added the same python path in the .cs file before I compiled the plugin)

Thanks in advance!

**Comment by rdeioris** [comment]
I fear the different versions of python are making mess. Unfortunately not having an embedded distribution for python2.7 is a real pain. Ensure you have a 64bit install of python and check the order of paths in the PATH env var. Eventually try to explicitely set python config like described here:  https://github.com/20tab/UnrealEnginePython#plugin-configuration

**Comment by kaedehao** [comment]
The issue is not bound to python version. Reason being python install path not found by UE. Only python27.dll found causes UE4 hangs at 73%. As long as python install path can be located, UE4 will launch successfully.

**Comment by lsr1994** [comment]
> The issue is not bound to python version. Reason being python install path not found by UE. Only python27.dll found causes UE4 hangs at 73%. As long as python install path can be located, UE4 will launch successfully.

Hey, man, I have the same problem. Have you solved it? Ask for help!

**Comment by kaedehao** [comment]
> > The issue is not bound to python version. Reason being python install path not found by UE. Only python27.dll found causes UE4 hangs at 73%. As long as python install path can be located, UE4 will launch successfully.
> 
> Hey, man, I have the same problem. Have you solved it? Ask for help!

Yes I've got it resolved. The built-in search path does not include \binary\win64\python27. So put python27 folder directly into win64 won't work. You'd have to package your main decencies in a way that the default search path can find them, for example, you can mimic how embedded python3 is structured. I'm not currently at my workstation, if you still have problem with this i'll show you what exactly i did.
